File: prob25/prob25.java
import java.math.BigInteger;

public class prob25 {
    public static void main(String[] args) 
    {
	BigInteger a= BigInteger.valueOf(1);
	BigInteger b = BigInteger.valueOf(1);
	BigInteger counter = BigInteger.valueOf(1);
	
	BigInteger max = (BigInteger.valueOf(10)).pow(999);
	while(true) {
	    BigInteger temp;
	    if(a.compareTo(max) != -1) {
		System.out.println("n=" + counter);
		return;
	    }
	    temp = a;
	    a= b;
	    b= temp.add(b);
	    counter = counter.add(BigInteger.ONE);
	}
    }
}

File: prob25/cpp/prob25.cpp
#include<iostream>
//
// broken. Must find a biginteger class first
//
int main() 
{
  long a=1;
  long b=1;
  long counter=1;
  while(true) {
    int temp;
    if(a >= 10 ^ 999) {
      std::cout << "n=" << counter << std::endl;
      return;
    }
    temp =a;
    a=b;
    b= temp+b;
    counter++;
  }

}
